Edit displaying-data page to introduce "Components and Templates" section (#35507)
angular/angular: aio/content/guide/displaying-data.md

# Displaying data in views

Angular [components](guide/glossary#component) form the data structure of your application.
The HTML [template](guide/glossary#template) associated with a component provides the means to display that data in the context of a web page.

> Together, a component and template form a [view](guide/glossary#view) of your application data.

Reviewer: ```suggestion
Together, a component's class and template form a [view](guide/glossary#view) of your application data.
```
Because technically it's a component class and a component template, rather than a component and a template.

The process of combining data values with their representation on the page is called [data binding](guide/glossary#data-binding).

> You display your data to a user (and collect data from the user) by *binding* controls in the HTML template to the data properties of the component.

Reviewer: ```suggestion
You display your data to a user (and collect data from the user) by *binding* controls in the HTML template to the data properties of the component class.
```

## Template inline or template file?

The `@Component` metadata tells Angular where to find the component's template.
You can store your component's template in one of two places.

* You can define the template *inline* using the `template` property of the `@Component` decorator.

> * You can define the template in a separate HTML file and link to that file in the `templateUrl` property of the `@Component` decorator.

Reviewer: ```suggestion
* Alternatively, you can define the template in a separate HTML file and link to that file in the `templateUrl` property of the `@Component` decorator.
```

Or something that introduces that these are two different ways of doing it.

> The choice between inline and separate HTML is a matter of taste, circumstances, and organization policy.

Reviewer: Generally, it has to do with the size of the template. If it's larger than a few lines, people usually  pull it out into its own file so that the file doesn't become cluttered. In real apps, this is almost a given (hence the CLI giving you a separate template by default) because even small experimental apps have a habit of growing quickly.

> ## Template inline or template file?

Reviewer 2: Nit: I don't like questions in headings. As we are changing this topic anyway, could we make this: "Choosing a template source"

Feel free to move this to a separate issue if needed.

The `@Component` metadata tells Angular where to find the component's template.
You can store your component's template in one of two places.

> * You can define the template *inline* using the `template` property of the `@Component` decorator. This is useful if the template is a small demo or test.

Reviewer 2: Change: "This is useful..." to "Inline templates are useful..."

Reason: Terms like "this" and "it" are notoriously ambiguous. This change makes the statement a bit clearer.

> * Alternatively, you can define the template in a separate HTML file and link to that file in the `templateUrl` property of the `@Component` decorator. This is the typical configuration for anything more complex than a small test or demo, and is the default when you generate a new component.

Reviewer 2: Same as above. Change: "This is the typical configuration..." to "This configuration is typical..."

  ng generate class hero
</code-example>

> This command creates the following code.

Reviewer 2: This is so much better! :D
